### PR TITLE
GDPR: Added checkbox and removed the dropdown from the login page

### PR DESCRIPTION
**Pull request by maximlion**
Title says pretty much everything, added tests as well.

Refs POS-2444

### CODE REVIEW

**Comment by maximlion** [comment]
Need to redo/add browser tests.

File: shuup/front/apps/auth/templates/shuup/user/login.jinja [hunk 1]
```diff
@@ -5,6 +5,12 @@
 
 {% block content_title %}{% endblock %}
 
+{# Removing the login dropdown from navigation #}
+{% block navigation %}
+        {{ render_navigation(quick_drop=False) }}
```
Comment (chessbr) [inline shuup/front/apps/auth/templates/shuup/user/login.jinja]: remove extra indentation

File: shuup/front/template_helpers/general.py [hunk 2]
```diff
@@ -24,6 +24,13 @@
 from shuup.front.utils.views import cache_product_things
 from shuup.utils.mptt import get_cached_trees
 from shuup.utils.translation import cache_translations_for_tree
+from shuup.front.apps.auth.forms import EmailAuthenticationForm
+
+
+@contextfunction
+def get_login_fields(context, request):
```
Comment (chessbr) [inline shuup/front/template_helpers/general.py]: `get_login_form` fits better as it returns a form.. is context even used?
```py
def get_login_form(request):
   return EmailAuthenticationForm(request=request)
```

File: shuup/front/templates/shuup/front/macros/navigation.jinja [hunk 3]
```diff
@@ -1,11 +1,12 @@
 {% extends "shuup/front/macros/theme/navigation.jinja" %}
+{% from "shuup/front/macros/general.jinja" import render_field with context %}
 
-{% macro render_navigation() %}
+{% macro render_navigation(quick_drop=True) %}
```
Comment (chessbr) [inline shuup/front/templates/shuup/front/macros/navigation.jinja]: I guess `quick_drop` name is not too intuitive. this should be a more explained parameter name, like `show_quick_login`

File: shuup/front/templates/shuup/front/macros/navigation.jinja [hunk 4]
```diff
@@ -55,25 +56,24 @@
             {% endif %}
         {% endcall %}
     {% else %}
+    {% if quick_drop %} {# Checking if the login dropdown should be removed #}
```
Comment (chessbr) [inline shuup/front/templates/shuup/front/macros/navigation.jinja]: replace `{% else %}` with `{% elif quick_drop %}`

File: shuup/front/templates/shuup/front/macros/navigation.jinja [hunk 5]
```diff
@@ -55,25 +56,24 @@
...
-                    <div class="form-group">
-                        <label for="username">{% trans %}Username or email address{% endtrans %}</label>
-                        <input type="text" name="username" class="form-control">
-                    </div>
-                    <div class="form-group">
-                        <label for="password">{% trans %}Password{% endtrans %}</label>
-                        <input type="password" name="password" class="form-control">
-                    </div>
+                    {% set form = shuup.general.get_login_fields(request) %}
+                    {% for field in form %}
```
Comment (chessbr) [inline shuup/front/templates/shuup/front/macros/navigation.jinja]: use the same login as `login.jinja` to render hidden fields first

```jinja
{% for f in form.hidden_fields() %}{{ f }}{% endfor %}
{% for f in form.visible_fields() %}{{ render_field(f, show_help_text=False) }}{% endfor %}
```

File: shuup/front/template_helpers/general.py [hunk 6]
```diff
@@ -24,6 +24,13 @@
...
+    form = EmailAuthenticationForm(request=request)
```
Comment (tulimaki) [inline shuup/front/template_helpers/general.py]: This shouldn't be hardcoded.

SHUUP_LOGIN_VIEW=“shuup.front.apps.auth.views:LoginView” Put this under `settings.py` under `shuup.front`. Then in templatetag `from shuup.utils.importing import cached_load` `login_class = cached_load("SHUUP_LOGIN_VIEW")`. Finally take the form from the view instead.

File: shuup/front/settings.py [hunk 7]
```diff
@@ -9,6 +9,13 @@
 #:
 #: This is the easiest way to customize the order creation process
 #: without having to override a single URL or touch the ``shuup.front`` code.
+
+
+#: Used to get the login form for login dropdown
```
Comment (chessbr) [inline shuup/front/settings.py]: Improve this text. And move this to the bottom or top of the code. You placed this in the middle of a existing help text :smile: Check above the comments, it describes the `SHUUP_BASKET_ORDER_CREATOR_SPEC` below.

File: shuup/front/template_helpers/general.py [hunk 8]
```diff
@@ -24,6 +24,13 @@
 from shuup.front.utils.views import cache_product_things
 from shuup.utils.mptt import get_cached_trees
 from shuup.utils.translation import cache_translations_for_tree
+from shuup.utils.importing import cached_load
+
+
+def get_login_fields(request):
```
Comment (chessbr) [inline shuup/front/template_helpers/general.py]: change the of the function. it doesn't return login fields.

File: shuup/front/templates/shuup/front/macros/navigation.jinja [hunk 9]
```diff
@@ -54,26 +55,23 @@
                 {{ _render_dropdown_item(url("shuup:logout"), "fa fa-sign-out fa-fw", _("Log out"), True) }}
             {% endif %}
         {% endcall %}
-    {% else %}
+    {% elif show_quick_login %} {# Checking if the login dropdown should be removed #}
```
Comment (chessbr) [inline shuup/front/templates/shuup/front/macros/navigation.jinja]: _Checking whether the login dropdown should be **shown**_

File: shuup/front/templates/shuup/front/macros/navigation.jinja [hunk 10]
```diff
@@ -31,7 +32,7 @@
     {% endif %}
 {% endmacro %}
 
-{% macro render_info_dropdown() %}
+{% macro render_info_dropdown(show_quick_login) %}
```
Comment (chessbr) [inline shuup/front/templates/shuup/front/macros/navigation.jinja]: make this param optional. we don't know if someone depends on this.

Comment (maximlion) [reply shuup/front/templates/shuup/front/macros/navigation.jinja]: fixed

File: shuup/front/settings.py (hunk 7 above)
Comment (maximlion) [reply shuup/front/settings.py]: Fixed

File: shuup/front/templates/shuup/front/macros/navigation.jinja (hunk 9 above)
Comment (maximlion) [reply shuup/front/templates/shuup/front/macros/navigation.jinja]: Fixed

File: shuup/front/template_helpers/general.py (hunk 8 above)
Comment (maximlion) [reply shuup/front/template_helpers/general.py]: Fixed

**Pull request merged**